pex/dist_metadata.py

```diff
@@ -130,8 +130,26 @@ def find_dist_info_file(
     # combo of `-`, `_`, and `.` as name component separators.
     project_name_pattern = re.sub(r"[-_.]+", "[-_.]+", project_name)
 
+    # The relevant PEP for versions is https://www.python.org/dev/peps/pep-0440 which does not allow
+    # a `-` in modern versions but also stipulates that all versions (legacy) must be handled. It
+    # turns out wheel normalizes `-` to `_` and Pip has had to deal with this:
+    #   https://github.com/pypa/pip/issues/1150
+    #
+    # We also deal with this, accepting either `-` or `_` when a `-` is expected.
+    version_pattern = re.sub(
```

Comment:
N.B.: An existing unit test and another IT that both started failing drove / forced this fix.